pydantic/config.py

```diff
@@ -33,11 +33,18 @@ class ConfigDict(TypedDict, total=False):
     title: str | None
     """The title for the generated JSON schema, defaults to the model's name"""
 
+    class_title_generator: Callable[[str], str] | None
+    """A callable that takes a class name and returns the title for it. Defaults to `None`."""
```

Comment:
we should be clear about whether this works for:
* pydantic models?
* dataclasses?
* typeddicts?

Reply:
I works for all model and model-like (pydantic dataclass, builtin dataclass, typeddict) types. `ConfigDict` can be applied to all of them so IMO it's a little redundant. Changing the name to `model_title_generator` makes this even more clear I feel.


I will add more documentation if you feel it's necessary.